packages/react-components/react-utilities/src/compose/slot.ts

```diff
@@ -0,0 +1,115 @@
+import { isValidElement } from 'react';
+import type {
+  AsIntrinsicElement,
+  SlotComponent,
+  SlotComponentMetadata,
+  SlotRenderFunction,
+  SlotShorthandValue,
+  UnknownSlotProps,
+} from './types';
+import { SLOT_COMPONENT_METADATA_SYMBOL } from './constants';
+import * as React from 'react';
+import { SlotPropsRecord } from './types';
+import { ExtractSlotProps } from './types';
+import { PropsWithoutRef } from './types';
+import { getNativeElementProps } from '../utils/getNativeElementProps';
+
+export type SlotOptions<Props extends UnknownSlotProps> = {
+  elementType?:
+    | React.ComponentType<Props>
+    | (Props extends AsIntrinsicElement<infer As> ? As : keyof JSX.IntrinsicElements);
+  defaultProps?: Partial<Props>;
+};
+
+/**
+ * Creates a slot for a v9 component
```

Comment:
```suggestion
 * Creates a slot for a component
```

Let's not use "v9" in naming as we will have to change it later